orangecontrib/spectroscopy/data.py

```diff
@@ -837,16 +839,24 @@ def read_spectra(self):
             YRr = np.arange(YR-1, -1, -1) # needed to have the same orientation as in Gwyddion
 
             X = X.reshape((meta["YRes"], meta["XRes"]) + (1,))
+            
+            global SET_flow
+            if SET_flow:    #True if from class NeaReader()
+                return X
+
             data = _spectra_from_image(X, [1], XRr, YRr)
 
             return data
 
 
 class NeaReader(FileFormat, SpectralFileFormat):
 
-    EXTENSIONS = (".nea", ".txt")
+    EXTENSIONS = (".nea", ".txt", '.gsf')
```

Comment:
This will simplify your code: if would be better if the new v3 reader is not implemented in the same class. That would simplify the read_spectra() function here.

So please restructure your code so that you make a new class called NeaReaderGSF which only contains your new reading code and leave the old NeaReader as it was before your changes.

This is better because v2 and v3 NeaReaders do not share any functionality or extensions. Therefore separating the code into multiple classes makes it more maintainable and prevents possible bugs with selection of the correct reader. That will avoid the try...excepts read_spectra.